src/exact/ExactMapper.cpp

```diff
@@ -562,21 +569,22 @@ void ExactMapper::coreMappingRoutine(const std::set<unsigned short>& qubitChoice
     // Allow only 1 y_k_pi to be true
     if (config.encoding == Encoding::Naive) {
         for (unsigned long k = 1; k < reducedLayerIndices.size(); ++k) {
-            expr onlyOne    = c.int_val(0);
+            auto onlyOne    = LogicTerm(0);
             piCount         = 0;
             internalPiCount = 0;
             do {
                 if (skipped_pi.count(piCount) == 0 || !config.enableSwapLimits) {
-                    onlyOne = onlyOne + ite(y[k - 1][internalPiCount], c.int_val(1), c.int_val(0));
+                    onlyOne = onlyOne + LogicTerm::ite(y[k - 1][internalPiCount], LogicTerm(1), LogicTerm(0));
                     ++internalPiCount;
                 }
                 ++piCount;
             } while (std::next_permutation(pi.begin(), pi.end()));
-            opt.add(onlyOne.simplify() == 1);
+            auto t = onlyOne == LogicTerm(1);
+            lb->assertFormula(t);
```

Comment:
why the temporary variable here?
```suggestion
            lb->assertFormula(onlyOne == LogicTerm(1));
```